packages/support/lib/fs.js

```diff
@@ -48,9 +48,23 @@ const fs = {
     return true;
   },
 
+  /**
+   * Returns `true` if `path` is executable
+   * @param {PathLike} path
+   * @returns {Promise<boolean>}
+   */
+  async isExecutable(path) {
+    try {
+      await fsPromises.access(path, constants.R_OK | constants.X_OK);
```

Comment:
is just X_OK not enough?

Reply:
afaik, no.  some executables must be read, e.g., shell scripts.  we could add an option to the method to _only_ check for the `x` bit in the future if necessary, but it _probably_ won't come up.